### DISCUSSION TITLE
Integrating two models in one snippet for creation

### DISCUSSION THREAD

**Discussion by Codevsun** [wagtail/wagtail#12191] [Q&A]
I have created a company model that has user abstract model as a foreign key inside it. when i convert the company model into snippet to allow the admin to create a new instance it offers me a drop down menu of the already created users. but i want to give the admin the ability to create a user alongside the company. i tried costume snippetviews with django form but its not working. please help

**Comment by Ganesh-403** [comment]
@Codevsun From what I can tell, the default snippet editing workflow isn't designed to create a new object for a `ForeignKey` field inline.

If your `Company` model has something like:

```python
class Company(models.Model):
    user = models.ForeignKey(User, on_delete=models.CASCADE)
```

then the snippet edit form will render `user` as a standard chooser/dropdown of existing users. That's the expected Django/Wagtail behavior.

If your goal is to create both a `User` and a `Company` in a single form, there isn't built-in support for that in the snippet editor. The snippet form is a regular `ModelForm`, whereas creating multiple related models together typically requires either:

- a custom view that coordinates multiple forms, or
- a parent/child relationship using `ClusterableModel` + `ParentalKey` + `InlinePanel` (which doesn't apply to a `ForeignKey` to Django's `User` model). :contentReference[oaicite:0]{index=0}

A few approaches you could consider:

1. **Recommended:** Create the `User` first, then select it when creating the `Company`. This follows the standard Django admin/Wagtail workflow.

2. **Custom admin view:** If creating both at once is a hard requirement, implement a custom create view containing both a `UserCreationForm` and a `CompanyForm`. After validating both, create the `User` first and then assign it to the `Company`.

3. **Separate profile model:** If the extra information logically belongs to the company rather than the authentication user, another common pattern is to keep `User` focused on authentication and store the business-specific data in your `Company` model linked via a `ForeignKey` or `OneToOneField`.

Could you share your model definitions for:

```python
class Company(...)
```

and your custom user model (or confirm whether you're using Django's default `User`)?

That would help determine whether a custom creation form is sufficient or whether the relationship itself could be modeled differently.

If this solves your problem, feel free to mark it as the accepted answer so others can find it easily.
